dev-python/cadquery-pywrap/cadquery-pywrap-7.4.0.ebuild

```diff
@@ -0,0 +1,73 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{8..9} )
+
+# The official pywrap 7.4.0 tarball is broken, but pywrap 7.5.0 has yet to be
+# officially released. We thus package a git commit known to work as expected.
+inherit git-r3 llvm toolchain-funcs distutils-r1
+
+MY_PN=occt
+MY_PV=$(ver_cut 1-2)
+MY_P="${MY_PN}${MY_PV}"
+
+DESCRIPTION="C++ binding generator based on libclang and pybind11"
+HOMEPAGE="https://github.com/CadQuery/pywrap"
+
+#FIXME: Uncomment on bumping to the next stable release and remove "git-r3".
+# SRC_URI="https://github.com/CadQuery/pywrap/archive/refs/tags/${MY_P}.tar.gz"
+EGIT_REPO_URI="https://github.com/CadQuery/pywrap.git"
+EGIT_BRANCH="master"
+EGIT_COMMIT="f8869e5a47fd3e3e1d31e7ab89b46c158f7487bf"
```

Comment:
```
COMMIT="f8869e5a47fd3e3e1d31e7ab89b46c158f7487bf"
SRC_URI="https://github.com/CadQuery/pywrap/archive/${COMMIT}.tar.gz"
```
And please rename the ebuild to `7.4.0_p<dateofcommit>` or `7.5.0_pre<dateofcommit>`

The thing with git-r3 is that it skips the Manifest and therefore we can't have any KEYWORDS.  (pkgcheck will not be happy if we put KEYWORDS in it anyway)

Reply:
> And please rename the ebuild to `7.4.0_p<dateofcommit>` or `7.5.0_pre<dateofcommit>`

**Absolutely.** I wasn't quite sure what to do there, so... I just did nothing. But let's go with `7.5.0_pre<dateofcommit>`, as that's pretty much what this is.

You're such a helpful code reviewer, by the way. Lovin' that constructive feedback!

> he thing with `git-r3` is that it skips the Manifest and therefore we can't have any KEYWORDS.

I... did not know that. In hindsight, that makes perfect sense. I'll just comment out the `KEYWORDS` line so that we can trivially restore that on the first bump. :+1: :+1:

Reply:
Sorry I explained a bit bad and too fast in my first reply. We actually can have KEYWORDS, but only if we do it like this (without git-r3):
```
COMMIT="f8869e5a47fd3e3e1d31e7ab89b46c158f7487bf"
SRC_URI="https://github.com/CadQuery/pywrap/archive/${COMMIT}.tar.gz"
```
Here we use GitHub's cool ability to turn any commit into a tarball, and because it is in SRC_URI now it will get hashed in the Manifest, which means that we can add the KEYWORDS. 

We can also do it like this (without any KEYWORDS) but the disadvantage is that things that DEPEND on this can't have KEYWORDS either.

Reply:
That's... *really* ingenious. Microsoft for the open-source win! Refactoring away `git-r3` is all upside and no downside, so I'll immediately make it so. And thanks as always for the educating heads-up; I always learn something exciting and dangerous when I submit PRs here. :full_moon_with_face: